src/torrent.cpp

```diff
@@ -7527,7 +7525,8 @@ namespace {
 		bdecode_node const metadata = bdecode(metadata_buf, ec, &pos, 200
 			, settings().get_int(settings_pack::metadata_token_limit));
 
-		if (ec || !m_torrent_file->parse_info_section(metadata, ec
```

Comment:
This causes https://github.com/qbittorrent/qBittorrent/issues/17595.
Previously, metadata merged into an existing `torrent_info` that contained some data from a magnet link (for example, trackers), so we just used it to (re)add a torrent.

Reply:
It could be fixed by replacing the following line
```c++
auto info = std::make_shared<torrent_info>(*m_torrent_file);
```
with the following one
```c++
auto info = std::make_shared<torrent_info>(*m_torrent_file);
```

Reply:
> It could be fixed by replacing the following line
> 
> ```c++
> auto info = std::make_shared<torrent_info>(*m_torrent_file);
> ```
> 
> with the following one
> 
> ```c++
> auto info = std::make_shared<torrent_info>(*m_torrent_file);
> ```

they are the same??

Reply:
> Did you mean `replace`
> 
> ```c++
> auto info = std::make_shared<torrent_info>(old_ih);
> ```

Exactly!
Thanks.

Reply:
the trackers part of `m_torrent_file` have (presumably) already been parsed into `torrent::m_trackers`, haven't they?
I'm hoping that any non-info-section fields can be removed from `torrent_info` in the long-run.

Reply:
I see a comment on line 332 suggesting the trackers are also saved in `m_torrent_file` in order to propagate to `create_torrent()`. Is that the path where you get the discrepancy?

Reply:
https://github.com/arvidn/libtorrent/pull/7031

Reply:
> I'm hoping that any non-info-section fields can be removed from `torrent_info` in the long-run.

Yes. I agree. I believe `torrent_info` should become an immutable representation of info section in future. But until then, such changes are still unexpected.

Reply:
> I see a comment on line 332 suggesting the trackers are also saved in `m_torrent_file` in order to propagate to `create_torrent()`. Is that the path where you get the discrepancy?

No. We use it differently.
When user adds magnet interactively we add it to libtorrent session behind the scenes in order to obtain metadata and allow user to select needed files. Once metadata is received we remove the torrent from libtorrent session and then add it using obtained metadata as a regular torrent file.

Reply:
Of course the scenario above will be adjusted when `torrent_info` becomes "info section" only.